cpp/ql/test/library-tests/dataflow/dataflow-tests/test.cpp

```diff
@@ -715,8 +715,22 @@ void write_to_star_buf(unsigned char *buf) // $ ast-def=buf
   write_to_star_star_p(&c);
 }
 
-void test(unsigned char *source) // $ ast-def=source
+void test_write_to_star_buf(unsigned char *source) // $ ast-def=source
 {
   write_to_star_buf(source);
   sink(*source); // clean
-}
\ No newline at end of file
+}
+
+void does_not_write_source_to_dereference(int *p) // $ ast-def=p ir-def=*p
+{
+  int x = source();
+  p = &x;
+  *p = 42;
+}
+
+void test_does_not_write_source_to_dereference()
+{
+  int x;
+  does_not_write_source_to_dereference(&x);
+  sink(x); // $ ast,ir=733:7 SPURIOUS: ast,ir=726:11
+}
```

Comment:
This was a testcase I came up with as an attempt to further reduce the testcase added in https://github.com/github/codeql/pull/13425/commits/90ffb45837728982ce6ff1614104546bb6a7f63a. However, it looks like this is a different conflation problem since test this wasn't fixed by this PR.